### ISSUE TITLE
kTLS detection in Configure doesn't work for cross compilation

### ISSUE THREAD

**Issue by 0140454** [openssl/openssl#19212]
**Version**

v3.0.5

**Description**

For crossing compilation, I executes the following command to configure OpenSSL.
```bash
./Configure \
	--cross-compile-prefix="x86_64-pc-linux-gnu-" \
	enable-ktls \
	shared
```

However, I found that kTLS detection doesn't work as expected, no matter what value I passed via `--cross-compile-prefix`.

After investigation, it seems that `Configure` uses a wrong config entry [`$config{cross_compile_prefix}`](https://github.com/openssl/openssl/blob/master/Configure#L1712-L1718).

```perl
        my $usr = "/usr/$config{cross_compile_prefix}";
        chop($usr);
        if ($config{cross_compile_prefix} eq "") {
            $usr = "/usr";
        }
        my $minver = (4 << 16) + (13 << 8) + 0;
        my @verstr = split(" ",`cat $usr/include/linux/version.h | grep LINUX_VERSION_CODE`);

        if ($verstr[2] < $minver) {
            disable('too-old-kernel', 'ktls');
        }
```

I think it should be `$config{CROSS_COMPILE}` ?

**Comment by 0140454** [comment]
By the way, I'm curious about that If my kernel header is in a different directory (e.g. `/usr/local/x86_64-pc-linux-gnu/sysroot/usr/include/linux/version.h`, not `/usr/x86_64-pc-linux-gnu/include/linux/version.h`)

Is there a recommended method to let Configure find it in the specified directory?

**Comment by t8m** [comment]
IMO replacing the detection with something similar to the freebsd routine would make sense - presence of linux/tls.h in the include path of the cross-compile environment should be good enough indicator.

**Comment by t8m** [comment]
Fix in #19213
